src/MongoDB.Driver.Core/Core/Clusters/DnsMonitor.cs

```diff
@@ -161,7 +161,7 @@ private void Monitor()
         {
             while (true)
             {
-                if (_cluster.ShouldDnsMonitorStop())
+                if (_processDnsResultHasEverBeenCalled && _cluster.ShouldDnsMonitorStop())
```

Comment:
Happy that this fix works, but was concerned why `ShouldDnsMonitorStop` was erroneously returning true.

Per off-line discussion with Dima, the cluster seems thinks it's a replicaset after the initial parsing of  `TXT` records due to the generated `clusterConnectionMode` option.
Dima discovered the following is called after TXT resolving: 
https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1199

https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Clusters/ClusterFactory.cs#L43

TLDR: Bug seems harmless with this fix.

Reply:
>was concerned why ShouldDnsMonitorStop was seemingly erroneously returning true.

no, it's not an error, `ShouldDnsMonitorStop`, the result of this method is correct. Per the spec, a replica set cluster type should not be processed via this logic.

> the cluster seems thinks it's a replicaset after the initial parsing of TXT records

the original connection string describes `replica set`. So, this's expected behavior

Reply:
The semantics of the method are now more like `ShouldDnsMonitorStopAfterFirstSuccessfulAttempt`